libs/blocks/table/table.css

```
@@ -456,6 +456,87 @@
   }
 }
 
+.table.has-addon .row-heading .col.col-heading {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  gap: var(--spacing-xxs);
+  flex: 1 1 0;
+  padding: var(--spacing-xs);
+}
+
+.table.has-addon p.body {
+  font-size: var(--type-body-xxs-size);
+  line-height: var(--type-body-xxs-lh);
+  margin-bottom: var(--spacing-xxs);
+}
+
+.table.has-addon .label {
+  background-color: var(--color-gray-300);
+  font-size: var(--type-body-xs-size);
+  line-height: var(--type-body-xs-lh);
+  border-radius: 4px;
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+  text-align: left;
```

Maybe this to account for RTL OOTB?
```suggestion
  text-align: start;
```

---

great find!